sys/aarch64/boot.c

```diff
@@ -199,6 +208,39 @@ __boot_text static paddr_t build_page_table(void) {
 
   l0[L0_INDEX(DMAP_BASE)] = (pde_t)l1d | L0_TABLE;
 
+#if KASAN /* Prepare KASAN shadow mappings */
+  size_t kasan_sanitzied_size =
+    SUPERPAGESIZE + roundup2(va - KASAN_MD_SANITIZED_START,
+                             SUPERPAGESIZE * KASAN_SHADOW_SCALE_SIZE);
+  size_t kasan_shadow_size = kasan_sanitzied_size / KASAN_SHADOW_SCALE_SIZE;
+  vaddr_t kasan_shadow_end = KASAN_MD_SHADOW_START + kasan_shadow_size;
+  va = KASAN_MD_SHADOW_START;
+  kasan_sanitized_end = KASAN_MD_SANITIZED_START + kasan_sanitzied_size;
```

Comment:
We could get rid of the `kasan_sanitized_end` intermediate global variable if we do this:
```suggestion
  /* XXX _kasan_sanitized_end is at a high address which is not mapped yet,
   * so we access it using its physical address instead. */
  *(vaddr_t *)AARCH64_PHYSADDR(&_kasan_sanitized_end) =
    KASAN_MD_SANITIZED_START + kasan_sanitized_size;
```